internal/command/machine/flaps_api/post.go

```diff
@@ -0,0 +1,189 @@
+		absPath, err := filepath.Abs(filepath.Join(state.WorkingDirectory(ctx), path))
+		if err != nil {
+			return err
+		}
+		path = absPath
+	}
+
+	file, err := os.Open(path)
+	if err != nil {
+		return fmt.Errorf("failed to open config file: %w", err)
+	}
+
+	defer func() {
+		if e := file.Close(); err == nil {
+			err = e
+		}
+	}()
+
+	if err = json.NewDecoder(file).Decode(input); err != nil {
+
+		return fmt.Errorf("failed parsing machine configuration: %w", err)
+	}
+
+	return nil
+}
+
+func runPost(ctx context.Context) error {
+	var (
+		appName  = app.NameFromContext(ctx)
+		client   = client.FromContext(ctx).API()
+		io       = iostreams.FromContext(ctx)
+		colorize = io.ColorScheme()
+		err      error
+		app      *api.AppCompact
+		input    api.LaunchMachineInput
+	)
+
+	if err = loadMachineJson(ctx, flag.FirstArg(ctx), &input); err != nil {
+		return err
+	}
+
+	if appName == "" {
+		appName = input.AppID
+	}
+
+	if appName == "" {
+		app, err = mach.CreateApp(ctx, "Running a machine without specifying an app will create one for you, is this what you want?", "", client)
+		if err != nil {
+			return err
+		}
+	} else {
+		app, err = client.GetAppCompact(ctx, appName)
+		if err != nil && strings.Contains(err.Error(), "Could not find App") {
+			app, err = mach.CreateApp(ctx, fmt.Sprintf("App '%s' does not exist, would you like to create it?", appName), appName, client)
+			if err != nil {
+				return err
+			}
+			if app == nil {
+				return nil
+			}
+		}
+		if err != nil {
+			return err
+		}
+	}
+	if app == nil {
+		return nil
+	}
+
+	flapsClient, err := flaps.New(ctx, app)
+	if err != nil {
+		return fmt.Errorf("could not make API client: %w", err)
+	}
+	ctx = flaps.NewContext(ctx, flapsClient)
+
+	if app.PlatformVersion == "nomad" {
+		return fmt.Errorf("the app %s uses an earlier version of the platform that does not support machines", app.Name)
+	}
+
+	input.AppID = app.Name
+	// Support a few command-line overrides, but this shouldn't be a clone of `machine run`
+	if name := flag.GetString(ctx, "name"); name != "" {
+		input.Name = name
+	}
+	if region := flag.GetString(ctx, "region"); region != "" {
+		input.Region = region
+	}
+
+	// now actually launch the machine
+	machine, err := flapsClient.Launch(ctx, input)
+	if err != nil {
+		return fmt.Errorf("could not launch machine: %w", err)
+	}
+
+	id, instanceID, state, privateIP := machine.ID, machine.InstanceID, machine.State, machine.PrivateIP
+
+	fmt.Fprintf(io.Out, "Success! A machine has been successfully launched in app %s, waiting for it to be started\n", app.Name)
+	fmt.Fprintf(io.Out, " Machine ID: %s\n", id)
+	fmt.Fprintf(io.Out, " Instance ID: %s\n", instanceID)
+	fmt.Fprintf(io.Out, " State: %s\n", state)
+
+	// wait for machine to be started
+	if err := mach.WaitForStartOrStop(ctx, machine, "start", time.Minute*5); err != nil {
+		return err
+	}
+
+	if !flag.GetDetach(ctx) {
+		fmt.Fprintln(io.Out, colorize.Green("==> "+"Monitoring health checks"))
```

Comment:
```suggestion
		fmt.Fprintln(io.Out, colorize.Green("==> Monitoring health checks"))
```

Reply:
yeah idk why the original run was written like that lol